RCTVideo.m

```diff
@@ -126,16 +155,26 @@ - (void)setSrc:(NSDictionary *)source
 
   [_player pause];
   [_playerLayer removeFromSuperlayer];
+  _playerLayer = nil;
+  [_playerViewController.view removeFromSuperview];
+  _playerViewController = nil;
+  [self removePlayerTimeObserver];
 
   _player = [AVPlayer playerWithPlayerItem:_playerItem];
   _player.actionAtItemEnd = AVPlayerActionAtItemEndNone;
 
-  _playerLayer = [AVPlayerLayer playerLayerWithPlayer:_player];
-  _playerLayer.frame = self.bounds;
-  _playerLayer.needsDisplayOnBoundsChange = YES;
-
-  [self.layer addSublayer:_playerLayer];
-  self.layer.needsDisplayOnBoundsChange = YES;
+  const Float64 progressUpdateIntervalMS = _progressUpdateInterval / 1000;
+  // CMTimeShow(CMTimeMakeWithSeconds(progressUpdateIntervalMS, NSEC_PER_SEC));
+    
+  // @see endScrubbing in AVPlayerDemoPlaybackViewController.m of https://developer.apple.com/library/ios/samplecode/AVPlayerDemo/Introduction/Intro.html
+  __weak RCTVideo *weakSelf = self;
+  _timeObserver = [_player addPeriodicTimeObserverForInterval:CMTimeMakeWithSeconds(progressUpdateIntervalMS, NSEC_PER_SEC) queue:NULL usingBlock:
+                     ^(CMTime time)
+                     {
+                         [weakSelf sendProgressUpdate];
+                     }];
```

Comment:
The `startProgressTimer` / `stopProgressTimer` seemed unnecessary. I find `addPeriodicTimeObserverForInterval` so much easier to use and clear.